### ISSUE TITLE
Type used on a private property is exported from generated Rust code

### ISSUE THREAD

**Issue by ultimaweapon** [slint-ui/slint#6880]
### Bug Description

If I declare an `enum` and use it on a private property the enum also exported from the generated Rust code. This will lead to compilation error if I have the same enum name but on a different module.

I put the reproducible code in https://github.com/ultimaweapon/slint-bug. There are two `Page` enum in the `ui/menu.slint` and `ui/wizard.slint`, which cause following errors:

```
     |
6    |      # [allow (dead_code)] # [derive (Default , Copy , Clone , PartialEq , Debug)] pub enum r#Page {
     |                                                                                    --------------- variant or associated item `Intro` not found for this enum
...
1485 |                     ) . apply_pin (_self) . get ()) == (r#Page :: r#Intro)) as bool)) as _ }
     |                                                                   ^^^^^^^ variant or associated item not found in `Page`


```

### Environment Details

- Slint Version: 1.8.0
- Platform/OS: Arch Linux with GNOME on Wayland
- Programming Language: Rust
- Backend/Renderer: Default

### Product Impact

I can workaround by using a different name but it is not convenience.

**Comment by ogoffart** [comment]
Indeed, we should filter a bit to limit ourselves with exported enum/struct,  and maybe the ones used in public properties for compatibility

**Comment by ogoffart** [comment]
I'll merge this issue with #9358
